jsx/Form.js

```diff
@@ -132,20 +109,53 @@ var FormElement = React.createClass({
       </form>
     );
   }
-});
+}
+
+FormElement.propTypes = {
+  name: PropTypes.string.isRequired,
+  id: PropTypes.string,
+  method: PropTypes.oneOf(['POST', 'GET']),
+  action: PropTypes.string,
+  class: PropTypes.string,
+  columns: PropTypes.number,
+  formElements: PropTypes.shape({
+    elementName: PropTypes.shape({
+      name: PropTypes.string,
+      type: PropTypes.string
+    })
+  }),
+  onSubmit: PropTypes.func,
+  onUserInput: PropTypes.func
+}
+
+FormElement.defaultProps = {
+  name: null,
+  id: '',
```

Comment:
i understood in our offline conversation that changing the default value of id might be outside the scope of this PR?

Reply:
Outside the scope of #3471, not of the current PR that refactors and cleans up the library.